estuary-cdk/estuary_cdk/__init__.py

```diff
@@ -152,6 +152,8 @@ def dump_all_tasks(signum, frame):
             for base in reversed(self.__class__.__bases__):
                 if exit := getattr(base, "_mixin_exit", None):
                     await exit(self, log)
+            if exit := getattr(self.__class__, "exit", None):
```

Comment:
what's this?

Reply:
@jgraettinger this allows connectors to specify a "cleanup" function in their class so they can clean up resources before exit

Reply:
A better way to do that would be to define a method an `exit` method on BaseConnector with a default implementation that's called.

But I would first check whether the resources to be cleaned up can be factored into a base class of my connector that's a subclass of `Mixin`, and then resource acquisition and cleanup can be done through the existing `_mixin_enter`/ `_mixin_exit` mechanism.

Reply:
An example mixin for an HTTP session: https://github.com/estuary/connectors/blob/main/estuary-cdk/estuary_cdk/http.py#L209-L222